Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,206 @@
+
+# TODO
+
+* `FiniteGaloisIntermediateField` should be a `ConditionallyCompleteLattice` but isn't proved yet.
+
+-/
+
+open CategoryTheory Opposite
+open scoped IntermediateField
+
+variable (k K : Type*) [Field k] [Field K] [Algebra k K]
+
+/-- The type of intermediate fields of `K/k` that are finite and Galois over `k` -/
+structure FiniteGaloisIntermediateField extends IntermediateField k K where
+  [finiteDimensional : FiniteDimensional k toIntermediateField]
+  [isGalois : IsGalois k toIntermediateField]
+
+namespace FiniteGaloisIntermediateField
+
+instance : Coe (FiniteGaloisIntermediateField k K) (IntermediateField k K) where
+  coe := toIntermediateField
+
+instance : CoeSort (FiniteGaloisIntermediateField k K) (Type _) where
+  coe L := L.toIntermediateField
+
+instance (L : FiniteGaloisIntermediateField k K) : FiniteDimensional k L :=
+  L.finiteDimensional
+
+instance (L : FiniteGaloisIntermediateField k K) : IsGalois k L := L.isGalois
+
+variable {k K}
+
+lemma val_injective : Function.Injective (toIntermediateField (k := k) (K := K)) := by
+  rintro ⟨⟩ ⟨⟩ eq
+  simpa only [mk.injEq] using eq
+
+/-- Turns the collection of finite Galois IntermediateFields of `K/k` into a lattice. -/
+
+instance (L₁ L₂ : IntermediateField k K) [IsGalois k L₁] [IsGalois k L₂] :
+    IsGalois k ↑(L₁ ⊔ L₂) where
+
+instance (L₁ L₂ : IntermediateField k K) [FiniteDimensional k L₁] :
+    FiniteDimensional k ↑(L₁ ⊓ L₂) :=
+  .of_injective (IntermediateField.inclusion inf_le_left).toLinearMap
+    (IntermediateField.inclusion inf_le_left).injective
+
+instance (L₁ L₂ : IntermediateField k K) [FiniteDimensional k L₂] :
+    FiniteDimensional k ↑(L₁ ⊓ L₂) :=
+  .of_injective (IntermediateField.inclusion inf_le_right).toLinearMap
+    (IntermediateField.inclusion inf_le_right).injective
+
+instance (L₁ L₂ : IntermediateField k K) [Algebra.IsSeparable k L₁] :
+    Algebra.IsSeparable k ↑(L₁ ⊓ L₂) :=
+  .of_algHom _ _ (IntermediateField.inclusion inf_le_left)
+
+instance (L₁ L₂ : IntermediateField k K) [Algebra.IsSeparable k L₂] :
+    Algebra.IsSeparable k ↑(L₁ ⊓ L₂) :=
+  .of_algHom _ _ (IntermediateField.inclusion inf_le_right)
+
+instance (L₁ L₂ : IntermediateField k K) [IsGalois k L₁] [IsGalois k L₂] :
+    IsGalois k ↑(L₁ ⊓ L₂) := {}
+
+instance : Sup (FiniteGaloisIntermediateField k K) where
+  sup L₁ L₂ := .mk <| L₁ ⊔ L₂
+
+instance : Inf (FiniteGaloisIntermediateField k K) where
+  inf L₁ L₂ := .mk <| L₁ ⊓ L₂
+
+instance : Lattice (FiniteGaloisIntermediateField k K) :=
+  val_injective.lattice _ (fun _ _ ↦ rfl) (fun _ _ ↦ rfl)
+
+instance : OrderBot (FiniteGaloisIntermediateField k K) where
+  bot := .mk ⊥
+  bot_le _ := bot_le (α := IntermediateField _ _)
+
+@[simp]
+lemma le_iff (L₁ L₂ : FiniteGaloisIntermediateField k K) :
+    L₁ ≤ L₂ ↔ L₁.toIntermediateField ≤ L₂.toIntermediateField :=
+  Iff.rfl
+
+variable (k) in
+/-- The finite Galois intermediate field obtained from the normal closure of adjoining a
```

Comment:
I am not sure to understand this doc. From what is it obtained?

Reply:
Stated in another way.

Reply:
Thanks!